oauth_dropins/bluesky.py

```diff
@@ -0,0 +1,76 @@
+from atproto.exceptions import UnauthorizedError
+from . import views, models
+from flask import request
+from google.cloud import ndb
+from .webutil.util import json_dumps
+import atproto
+
+class Start(views.Start):
+  """
+  Starts Bluesky auth - used just to provide the button.
+  """
+  NAME = 'bluesky'
+  LABEL = 'Bluesky'
+
+class BlueskyAuth(models.BaseAuth):
+  """
+  An authenticated Bluesky user.
+  """
+
+  did = ndb.StringProperty(required=True)
+  password = ndb.StringProperty(required=True)
+  user_json = ndb.TextProperty(required=True)
+
+  def site_name(self):
+    return 'Bluesky'
+
+  def access_token(self):
+    """
+    Returns did and password as a tuple in place of an OAuth key id/secret pair.
+    """
+    return (self.did, self.password)
+
+  def _api(self):
+    """
+    Returns an atproto.Client.
+    """
+    (client, _) = BlueskyAuth._api_from_password(self.did, self.password)
+    return client
+
+  @staticmethod
+  def _api_from_password(username, password):
```

Comment:
minor nit, `s/username/handle` here and below?

Reply:
sure, think I just copied this from mastodon